src/main/java/com/iota/iri/service/Feature.java

```diff
@@ -0,0 +1,112 @@
+package com.iota.iri.service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.iota.iri.Iota;
+import com.iota.iri.conf.IotaConfig;
+
+/**
+ * 
+ * Features describe the capability of the node. 
+ * They do not have to have a name specifically linked to a class/action. 
+ * Features are based on enabled/disabled functionality due to, for example, config values
+ *
+ */
+public enum Feature {
+    /**
+     * This node allows doing proof of work for you.
+     */
+    PROOF_OF_WORK("RemotePOW"),
+    
+    /**
+     * This node has enabled snapshot pruning
+     * It will most likely just contain "recent" transactions, as defined in the nodes config
+     */
+    SNAPSHOT_PRUNING("snapshotPruning"),
+    
+    /**
+     * This node automatically tries to get the new IP from neighbors with dynamic IPs
+     * 
+     */
+    DNS_REFRESHER("dnsRefresher"),
+    
+    /**
+     * This node is connected to the testnet instead of the mainnet tangle
+     */
+    TESTNET("testnet"),
+    
+    /**
+     * This node has the zero message queue enabled for fetching/reading "activities" on the node
+     */
+    ZMQ("zeroMessageQueue");
+    
+    private String name;
+
+    Feature(String name) {
+        this.name = name;
+    }
+    
+    @Override
+    public String toString() {
+        return name;
+    }
+    
+    /**
+     * Calculates all features for this Iota node
+     * @param instance the instance of this node
+     * @return A list of features
+     */
+    public static Feature[] calculateFeatures(Iota instance) {
```

Comment:
Can this receive `IotaConfig`?

Reply:
It could be, but maybe some day we have to calculate features based on external factors outside of the config. Like only enable X if the database is up to date.

Or want to change to that when we have the need for it, and use config for now?

Reply:
use configs for now
global scope is evil